### DISCUSSION TITLE
Stuck on Thinking

### DISCUSSION THREAD

**Discussion by gregorysha** [TransformerOptimus/SuperAGI#647] [Q&A]
Hello, I have successfully installed and configured superagi locally, I've setup the google keys, openapi key, pinecone keys, I have kicked off a couple runs using supercoder but both get stuck on "thinking".  Visual Code shows activity but no progress has been made after 22 hours.  Any suggestions on troubleshooting?

![superagi](https://github.com/TransformerOptimus/SuperAGI/assets/75952552/25e88a4b-d684-468e-abe3-a7b3ee3c1a68)

**Comment by rhw2008** [comment]
Hello
I have the same issue. IS it because I have added 14 goals ? 
thanks

**Comment by neph1** [comment]
In my (limited) experience, if it's stuck on "thinking", it's a problem with the model, or there's an error in the logs. If it's been running for 22 hours the error is probably much higher up. Try running it again and see if it fails instantly. If there are no issues, you should probably get an answer within seconds (or at least minutes)

**Reply by joshknopp** [reply]
I failed to setup OpenAI as a paid account at first. The console was helpfully telling me what was wrong, but I was staring at the UI wondering what wasn't working. Something to double check for novices like me.

**Comment by neelayan7** [comment]
@gregorysha Is the issue still persisting?
@neph1 is correct. Please try that and let us know.

**Comment by rhw2008** [comment]
It's still persisting.  I've recreated the agent and restarted it several times, but nothing worked 
![screenshot_20230719_071414](https://github.com/TransformerOptimus/SuperAGI/assets/6476242/6d858eab-dd5d-4ac2-9d5b-8369d0ecf596)

**Comment by neelayan7** [comment]
Can you take the latest pull of main and try again?

**Reply by rhw2008** [reply]
I'm using the latest version that I built yesterday. What else should I do ?

**Reply by neelayan7** [reply]
Can you check the logs? Do you see any errors?
Also, have you put in the API keys of the tools you're using?

**Reply by rhw2008** [reply]
I'm using API for the tools. I’m using PINECONE_API_KEY / OPENAI_API_KEY / GOOGLE_API_KEY. I don't see any logs file in the folder "SuperAGI-main". Can you please tell where I can find them?

**Reply by Maximilian50001** [reply]
once you run "docker-compose up --build" in command prompt and it builds your app. Any UI runs will show as logs in the command prompt window. 

I am having celery1 "forkpool" errors saying I have exceeded openai rate limits. This happens on the very first API call. odd. I am monitoring openai usage and its fine meaning I am not hitting my account limits.

I have tried using superagi app call rates as high as 10,000 milisecs. 1 call every 10 seconds and still the rate limit error. When my bot hits rate limit it is always stuck "thinking".

Let me know if you find a solution. I am trying hard to resolve.
